### PR TITLE
IllustratedMessage component

### PR DESCRIPTION
**Pull request by victortrinh2**
# Adding the IllustratedMessage to Hopper

Issue: https://workleap.atlassian.net/browse/DS-207

## Description

Had to add the Image & SvgImage also to the make the component work properly.

## How to test

Check out the storybook and the documentation.

### Image
Screenshot 2024-12-04 at 5 47 37 PM

### SvgImage
Screenshot 2024-12-04 at 5 48 51 PM

### IllustratedMessage
image

### CODE REVIEW

File: packages/components/src/IllustratedMessage/assets/nasa.png
Comment (alexasselin008) [inline packages/components/src/IllustratedMessage/assets/nasa.png:1]: Can you find another image or use one of the frog image instead? 

Everything space/nasa related was the theme of "Orbit", but doesn't really fit in this new frog theme DS

File: packages/components/src/Image/tests/jest/SvgImage.tsx
Comment (alexasselin008) [inline packages/components/src/Image/tests/jest/SvgImage.tsx:1]: file name is wrong, it's usually `SvgImage.ssr.test.tsx`

File: packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+const meta = {
+    title: "Components/SvgImage",
+    component: SvgImage,
+    parameters: {
+        chromatic: {
+            delay: 100
+        }
+    }
+} as Meta<typeof SvgImage>;
```
Comment (alexasselin008) [inline packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx:16]: `as Meta` should instead be `satisfies Meta`

There is a slack thread that details why this is relevant here: https://workleap.slack.com/archives/C05LE4QPHKL/p1708107373983879

File: packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Stroke = {
+    render: () => (
+        <SvgImage stroke="neutral" src={NoResults} aria-label="No Results" />
+    )
+} as Story;
```
Comment (alexasselin008) [inline packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx:26]: `satisfies Story` also for the same reason

File: packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    }
+} as Meta<typeof SvgImage>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Stroke = {
+    render: () => (
```
Comment (alexasselin008) [inline packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx:23]: prefer the args synthax over the render function whenever possible


```
const meta = {
    title: "Components/SvgImage",
    component: SvgImage,
    parameters: {
        chromatic: {
            delay: 100
        }
    },
    args: {   <------------------ args on all stories
        src: NoResults,
        aria-label: "No Results"
    }
} satisfies Meta<typeof SvgImage>;


export const Stroke = {
    args: {
        stroke: "neutral"
    }
} satisfies Story;

export const Styling = {
    render: (args) => (   <-- This is a good place to use render
        <Inline>
            <SvgImage className="stroke-red" {...args} />
            <SvgImage style={{ stroke: "red" }} {...args} />
        </Inline>
    )
} satisfies Story;
```

Comment (alexasselin008) [reply packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx:23]: I won't comment on every .stories.tsx, but i would expect this for all the files you created please

File: packages/components/src/Image/src/Image.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+     * The image shape.
+     */
+    shape?: "straight" | "rounded" | "circular";
+    /**
+     * An image path.
+     */
+    src?: ResponsiveProp<string>;
+}
+
+function Image(props: ImageProps, ref: ForwardedRef<HTMLImageElement>) {
```
Comment (alexasselin008) [inline packages/components/src/Image/src/Image.tsx:29]: i saw this in Spectrum, and it's pretty smart. 

```

  if (!alt) {
    console.warn(
      'The `alt` prop was not provided to an image. ' +
      'Add `alt` text for screen readers, or set `alt=""` prop to indicate that the image ' +
      'is decorative or redundant with displayed text and should not be announced by screen readers.'
    );
  }
```

Comment (alexasselin008) [reply packages/components/src/Image/src/Image.tsx:29]: what do you think?

File: packages/components/src/Image/docs/size.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Image, Inline } from "@hopper-ui/components";
+
+export default function Example() {
+    return (
+        <Inline>
+            <Image src="/frog.jpg" alt="Frog" width="core_640" height="auto" />
```
Comment (alexasselin008) [inline packages/components/src/Image/docs/size.tsx:6]: Maybe we should show examples with the UNSAFE_width instead? most of the time, images aren't the size of tokens anyway. The main use case will be UNSAFE_width="1174px" most likely

File: packages/components/src/Image/tests/jest/Image.test.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    });
+
+    it("should support slots", () => {
+        render(
+            <ImageContext.Provider value={{ slots: { test: { "aria-label": "test" } } }}>
+                <Image data-testid="Image" slot="test" src="https://i.pravatar.cc/96?img=1" />
+            </ImageContext.Provider>
+        );
+
+        const element = screen.getByTestId("Image");
```
Comment (alexasselin008) [inline packages/components/src/Image/tests/jest/Image.test.tsx:44]: i know there serie of tests is copy pasted so no worry, just an FYI, i always refer to this list of priority of what to query: https://testing-library.com/docs/queries/about/#priority

getByTestId should always be the "last resort"

File: packages/components/src/Image/src/Image.tsx (hunk 4 above)
Comment (victortrinh2) [reply packages/components/src/Image/src/Image.tsx:29]: Yeah good idea!

File: packages/components/src/Image/docs/size.tsx (hunk 5 above)
Comment (victortrinh2) [reply packages/components/src/Image/docs/size.tsx:6]: Feels weird to expose that we want to use something UNSAFE in the documentation. But I get it for this case.

File: packages/components/src/IllustratedMessage/src/IllustratedMessage.module.css [hunk 7]
```diff
@@ -0,0 +1,38 @@
+.hop-IllustratedMessage {
+    display: grid;
```
Comment (alexasselin008) [inline packages/components/src/IllustratedMessage/src/IllustratedMessage.module.css:2]: we try to create a third level of tokens, which is the "component tokens" inside the components. For everything that is relevant.

For instance, in this example
```
    --hop-IllustratedMessage-sm-size: 20rem;
    --hop-IllustratedMessage-md-size: 25rem;
    --hop-IllustratedMessage-lg-size: 28.5rem;


.hop-IllustratedMessage--sm {
    --max-width: var(--hop-IllustratedMessage-sm-size);
}
```

At some point, we could extract --hop-IllustratedMessage-sm-size in the tokens packages, so people that are using JS framework other than react could use those tokens to re-create our components.

--max-width is considered an "internal variable" and is used when the value is reassigned a couple of time. Those are not going to be exported

File: packages/components/src/Image/tests/chromatic/Image.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+        alt: "Frog"
+    }
+} satisfies Meta<typeof Image>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Default = {
+    render: args => (
```
Comment (alexasselin008) [inline packages/components/src/Image/tests/chromatic/Image.stories.tsx:22]: You can remove this, this is the default implementation

File: packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        "aria-label": "No Results"
+    }
+} satisfies Meta<typeof SvgImage>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const Stroke = {
+    render: args => (
```
Comment (alexasselin008) [inline packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx:22]: export const Stroke = {
    args:{
        stroke: "neutral"
    }
} satisfies Story;

Comment (alexasselin008) [reply packages/components/src/Image/tests/chromatic/SvgImage.stories.tsx:22]: you don't need render function everywhere

File: packages/components/src/Image/tests/chromatic/Image.stories.tsx (hunk 8 above)
Comment (alexasselin008) [reply packages/components/src/Image/tests/chromatic/Image.stories.tsx:22]: export const Default = {
} satisfies Story;

File: packages/components/src/IllustratedMessage/docs/buttons.tsx [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import { Button, ButtonGroup, Content, Heading, IllustratedMessage, Stack, SvgImage } from "@hopper-ui/components";
+
+import { NoResults } from "../assets/index.ts";
+
+export default function Example() {
+    return (
+        <Stack>
```
Comment (alexasselin008) [inline packages/components/src/IllustratedMessage/docs/buttons.tsx:7]: can you add a small gap here, they are too close to each other

File: packages/components/src/IllustratedMessage/docs/image.tsx [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import { Content, Heading, IllustratedMessage, Image } from "@hopper-ui/components";
+
+export default function Example() {
+    return (
+        <IllustratedMessage>
+            <Image src="/frog.jpg" aria-label="No Results" />
```
Comment (alexasselin008) [inline packages/components/src/IllustratedMessage/docs/image.tsx:6]: For Images, they need a alt tag, not an aria label. Make sure to correct this everywhere

File: packages/components/src/IllustratedMessage/docs/buttons.tsx (hunk 10 above)
Comment (alexasselin008) [reply packages/components/src/IllustratedMessage/docs/buttons.tsx:7]: In fact never, can you create 2 different sample, and put them back to back

Comment (alexasselin008) [reply packages/components/src/IllustratedMessage/docs/buttons.tsx:7]: like in this section https://wl-orbiter-website.netlify.app/?path=/docs/card--docs

under buttons part

File: packages/components/src/IllustratedMessage/docs/image.tsx [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import { Content, Heading, IllustratedMessage, Image } from "@hopper-ui/components";
+
+export default function Example() {
+    return (
+        <IllustratedMessage>
```
Comment (alexasselin008) [inline packages/components/src/IllustratedMessage/docs/image.tsx:5]: In the Image doc, maybe add objectFit like this? https://react-spectrum.adobe.com/react-spectrum/Image.html#object-fit

**Review by alexasselin008 — APPROVED** [review]

**Pull request merged**